### PR TITLE
Add itemnames and count to chest log messages

### PR DESCRIPTION
**Pull request by rubenwardy**
Low priority enhancement.

### CODE REVIEW

**Comment by PilzAdam** [comment]
:+1:

**Comment by paramat** [comment]
Is it needed? It makes things slower and affects lua lag.

**Comment by kaeza** [comment]
>Is it needed? It makes things slower and affects lua lag.

Yeah. It's slower by probably several nanoseconds every time a player moves an item.  Probably adding up to tens of milliseconds over a whole server session. Stop making things slow rubenwardy!

:+1:

File: mods/default/nodes.lua [hunk 1]
```diff
@@ -1388,11 +1388,13 @@ minetest.register_node("default:chest", {
 	end,
     on_metadata_inventory_put = function(pos, listname, index, stack, player)
 		minetest.log("action", player:get_player_name() ..
-			" moves stuff to chest at " .. minetest.pos_to_string(pos))
+			" moves " .. stack:get_name() .. " " .. stack:get_count() ..
```
Comment (kaeza) [inline mods/default/nodes.lua]: Wouldn't it be better to use [`stack:to_string()`](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L2733)?

**Comment by paramat** [comment]
Sure i know it's not much :) but lua is slow and causes server lag. I can't see this as much needed, it's enough to know someone has used a chest at a certain location, item and number of items seems overkill. Maybe someone can convince me otherwise?

**Comment by kaeza** [comment]
Trading a (truly) negligible performance hit for more useful information in the logs? Is that not enough for you?

>lua is slow and causes server lag

Again, are you saying that milliseconds, in extreme cases a few seconds if even that much, over a whole server session, probably spanning 24 hours or more, really matters?

We are not talking about a function called 1 million times every server step. We are talking about a function that is called only when a player moves stuff in a chest, maybe once or twic per second at most. And you really say --with a straight face-- that this matters at all? Pardon my french, but what the actual fuck?

**Comment by kilbith** [comment]
Same reaction than @kaeza, what's the fuck ? Seriously I think you're inadequate for discussing efficient code outside mapgens (though your vigilance is appreciated), as you obviously lack of knowledge among the best modders around. Not an offense, just a revealed fact.

> but lua is slow and causes server lag. I can't see this as much needed, it's enough to know someone has used a chest at a certain location, item and number of items seems overkill. 

LuaJIT can be (almost) as fast as C, and there are a very few datas to pass here.

Slowness of moving stuff ain't the fault of Lua, but the laborious manner the engine deals with metadatas and collateral consequences (like numerous mapblock updates). How big is the exploitation of metadatas in a callback doesn't matter much, it's the fact of engaging such a callback that counts.

As an experienced modder with formspec/metas, I highly doubt that this adds more than 1-2 ms of processing. Totally negligible if you realize what it is.

**Comment by paramat** [comment]
I mostly agree. I'm not talking about slowness to move stuff, but adding another small amount of lua server lag. I can't see the usefulness of the extra information but am happy to be convinced otherwise and expect i will be.

The smallness of the processing time is irrelevant, if something is not needed it should not be merged for simplicity, and because however small, these amounts of time add up eventually. A less careful attitude results in code bloated with useless features and a slower game. I'm sure you know this so why the surprised reaction? It's your attitudes that are sloppy here.

So please can players explain why this is needed? Why would the amount of items be important? The type of item seems more important, if needed at all.

**Comment by kilbith** [comment]
> In code 1-2ms is a huge amount of time

Is it a joke or something ? This is not a critical code here, and the cause of the formspec's slowness or lag is rooted somewhere else. I'm just speechless to read this.

Also I was in the `#lua` IRC channel, some expert found out your comment. I leave you observe the reputation you arouse on us and the project : 

```
<zash> TIL LuaJIT is too slow for games
<zash> https://github.com/minetest/minetest_game/pull/768#issuecomment-168527989 what
<Neffi> "it can't be used by many due to the OOM crashes it causes"
<Neffi> I wonder how shit their other code is
<Neffi> also why are you even reading minecraft mod stuff. it is all awful 
<xyzzy> luaSHIT...
<xyzzy> although i'd kinda be interested in how they're actually OOMing in lua
<Neffi> Is it better because you can more easily see how shit it is? 
<zash> Doesn't even require the latest in gaming computers to show a bunch of cubes.
<xyzzy> sure luajit has a stupid low memory limit but that's still like a gig or two
<xyzzy> is that log like... keeping strings forever or something
<Neffi> LUA LAG xyzzy
<Neffi> it LAGS so much
<Neffi> the language, not their code
```

**Comment by paramat** [comment]
> I think you're inadequate for discussing efficient code outside mapgens (though your vigilance is appreciated), as you obviously lack of knowledge

In my earlier comment i made it clear i know it's a tiny amount of processing, your argument fails.
>  This is not a critical code here, and the cause of the formspec's slowness or lag is rooted somewhere else

You miss the point, i'm not concerned about the tiny slowdown (experienced by the player involved) of moving the items or the formspec, the point is any extra load on the lua server creates lag elsewhere for all players. How tiny the extra lag is is also not the main point. You consistently misunderstand, try reading what i wrote.
It is obvious that 1-2ms is not insignificant in code.
> <zash> TIL LuaJIT is too slow for games

A misunderstanding of the issue, see below.
> <xyzzy> sure luajit has a stupid low memory limit but that's still like a gig or two

This is what i'm referring to, we have mods (complex lua mapgens) that use 2+GB of memory due to use of noise and are causing OOM crashes for users, kilbith i know you follow MT closely so you will be aware of this. This is forcing some players to not use LuaJIT.

The discussion you posted is just people misunderstanding the issue and making some harmless harsh comments in ignorance of the details (note someone said minecraft), don't worry about it.
I wonder how they 'found' the comment?

There's a reason the waterlily is placed as a single-node schematic, simple decorations cannot have random rotation, they are also placed in deep water which is unsuitable.

**Comment by kaeza** [comment]
>A misunderstanding of the issue, [...]

No. Excuse me, but you are now changing the topic. Here, I'll quote your original comment:

>[...] It makes things slower and affects lua lag.

I won't repeat what I already said two times.

To answer to your (new) argument about OOM issues: Do you have any proof that this affects it at all? Otherwise, I again call bullshit on your argument.

What I can agree on is that maybe the count/wear may be unneeded, but neither because of performance nor memory reasons; because it may not be so useful (but this is up for discussion).

Blocking this PR based on these arguments is, frankly, idiotic.

**Comment by kilbith** [comment]
> It is obvious that 1-2ms is not insignificant in code.

1-2 ms is usually a "statistical noise" in mods benchmarking. When you profile some Lua code, this is randomly ~1-2 ms less or more on each test. >10 ms of difference in a _mod profiling_ is relevant diagnostically wise, not 1-2 ms.

There are many areas where I could save a bunch of milliseconds in minetest_game, but you wouldn't like that because the readability would be ugly and it's not worth it (server steps are much slower than metadatas handling, no matter what mods you use).

**Comment by sofar** [comment]
I can't envision any significant performance degradation from this patch. The messages are already getting printed, string concatenation is cheap, and the expensive pos_to_string() is already there. The get_count and get_name lookups also should be cheap.

:+1:  from me

**Comment by paramat** [comment]
kaeza, maybe it's a language barrier but like kilbith did you are continuing to misunderstand and are getting ridiculously worked up and aggressive over this, you come over like a raging madman and i am calmly trying to repeatedly explain.
> To answer to your (new) argument about OOM issues: Do you have any proof that this affects it at all?

Read the forums and IRC logs, you will see there are several people having problems recently with OOM crashes, mostly resulting from LuaJIT having a lower memory limit. The only reason i mention this is because earlier the argument was made that the (tiny) amount of extra processing doesn't matter because LuaJIT is fast, i am pointing out that we can't assume that everyone uses LuaJIT, many people do not, even some devs, like me. The project cannot rely on LuaJIT.

I am not blocking this i am actually neutral, other devs will probably approve.

I can see a possible usefulness for item kind, but not item count, perhaps we should consider this instead.

Note that, still, no-one has actually explained why this is useful or needed. even the author labels it as low priority.

I'll explain again since you continue to ignore this, even if this commit added no more processing, i would still ask is it really needed? In the interests of simplicity. The raging responses to my reasonable question are astonishing. Now chill out.

**Comment by kilbith** [comment]
> The project cannot rely on LuaJIT.

I'd say : abusively greedy mods cannot rely on the project, instead. But that's another topic.

**Comment by ShadowNinja** [comment]
:+1: I think kaeza's comment (using `:to_string`) should be addressed though.

**Comment by paramat** [comment]
> What I can agree on is that maybe the count/wear may be unneeded

We should consider adding item type only, not quantity.

**Comment by paramat** [comment]
If this is changed to notify of item only, not item count, i will +1 and merge it.

**Comment by paramat** [comment]
Just needs commit message updating, otherwise +1.

**Comment by paramat** [comment]
:+1: WIll merge soon.

**Comment by paramat** [comment]
42cc6d2c311d4b70f328d3e31554bf90dd9fc45e

**Pull request closed without merging**